sdk/typescript/runtime/main.go

```diff
@@ -400,26 +406,34 @@ func (t *TypescriptSdk) detectRuntime(ctx context.Context) (SupportedTSRuntime,
 
 			switch runtime := SupportedTSRuntime(runtime); runtime {
 			case Bun, Node:
-				return runtime, version, nil
+				t.moduleConfig.runtime = runtime
+				t.moduleConfig.runtimeVersion = version
+
+				return nil
+			// Custom base image different from node or bun
 			default:
-				return "", "", fmt.Errorf("detected unknown runtime: %s", runtime)
+				t.moduleConfig.runtime = Node
+				t.moduleConfig.image = value
 			}
 		}
 	}
 
 	// Try to detect runtime from lock files
 	if t.moduleConfig.hasFile("bun.lockb") {
-		return Bun, "", nil
+		t.moduleConfig.runtime = Bun
+
+		return nil
 	}
```

Comment:
What I'm seeing is that if the user provides an image ref with a digest to `dagger.runtime` but has a `bun.lockb`, the final image will be:

```
oven/bun:sha256:xxxxxxxxxxxxxxxxx-alpine
```

Reply:
No, because I return before in case there's an explicit config from the user

https://github.com/TomChv/dagger/blob/feat/custom-ts-sdk-runtime-base-image/sdk/typescript/runtime/main.go#L412